nexus/db-model/src/bytecount.rs

```diff
@@ -63,3 +64,155 @@ impl From<BlockSize> for ByteCount {
         Self(bs.to_bytes().into())
     }
 }
+
+impl From<i64> for ByteCount {
+    fn from(i: i64) -> Self {
+        i.into()
+    }
+}
+
+impl From<ByteCount> for i64 {
+    fn from(b: ByteCount) -> Self {
+        b.0.into()
+    }
+}
+
+impl TryFrom<diesel::pg::data_types::PgNumeric> for ByteCount {
+    type Error = anyhow::Error;
```

Comment:
Is there a reason to do the conversion to/from the `PgNumeric` type specifically? Why not the https://docs.diesel.rs/master/diesel/sql_types/struct.Numeric.html type? (coping with individual digits at a time seems like a pain in the butt)

Reply:
It is a pain, but I don't see a way to chain `Into` or `From` here to go from either `PgNumeric` or `Numeric` into i64.

Reply:
Taking a step back - I don't just mean this function. I mean, why are we using `PgNumeric` in the db? There are other DB types which could be used instead here, like https://docs.diesel.rs/master/diesel/sql_types/struct.BigInt.html , if you're just going to / from i64.

Reply:
We're not using `PgNumeric` in the db - as far as I understand, `diesel::sql_types::Numeric` is being returned due to the use of the sum. Trying to use any other type, I kept running into errors like

```
error[E0277]: the trait bound `BigInt: FromSql<diesel::sql_types::Numeric, Pg>` is not satisfied
    --> nexus/src/db/datastore/region.rs:214:26
     |
214  |                         .get_result(conn)?;
     |                          ^^^^^^^^^^ the trait `FromSql<diesel::sql_types::Numeric, Pg>` is not implemented for `BigInt`
     |
```

and eventually just went with `PgNumeric` because I could get at the enum variants to actually turn it into `ByteCount`.

Reply:
- The usage of Numeric, or the usage of PgNumeric, to me implies "this is a variable precision number, **not** an integer"
- The usage of `smallint` / `integer` / `bigint` implies "this is an integer, of a particular size"

"byte count" seems like it should be an integer, not a variable-precision number, right?

Reply:
Also, AFAICT:

- [`sum`](https://docs.diesel.rs/master/diesel/dsl/fn.sum.html) acts on types which are `Foldable`
- [`Foldable`](https://docs.diesel.rs/master/diesel/sql_types/trait.Foldable.html) is implemented for the integer types: https://docs.diesel.rs/master/diesel/sql_types/trait.Foldable.html#impl-Foldable-for-BigInt

Reply:
Right but if I follow the Foldable link to the source, I see

```rust
foldable_impls! {
    ...
    sql_types::BigInt => (sql_types::Numeric, sql_types::Numeric),
    ...
}
```

which I think means that the sum widens the type from BigInt to Numeric?

Reply:
got it. Okay, don't hold up the PR on this point. I think we're in agreement that "it would be nice if bytes could be integer types", but if the implicit SQL conversions make that difficult, we can punt on it.

(But dang, would be nice if we could coerce them to stay as integer-based types!)